src/flows/pipes/QueryBuilder/context.tsx

```diff
@@ -363,21 +376,35 @@ export const QueryBuilderProvider: FC = ({children}) => {
       ? EXTENDED_TAG_LIMIT
       : DEFAULT_TAG_LIMIT
 
-    // TODO: Use the `v1.tagValues` function from the Flux standard library once
-    // this issue is resolved: https://github.com/influxdata/flux/issues/1071
-    query(
-      `${_source}
-              |> range(${formatTimeRangeArguments(range)})
-              |> filter(fn: (r) => ${tagString})
-              |> keep(columns: ["${cards[idx].keys.selected[0]}"])
-              |> group()
-              |> distinct(column: "${
-                cards[idx].keys.selected[0]
-              }")${searchString}
-              |> limit(n: ${limit})
-              |> sort()`,
-      scope
-    )
+    let queryText = `${_source}
+    |> range(${formatTimeRangeArguments(range)})
+    |> filter(fn: (r) => ${tagString})
+    |> keep(columns: ["${cards[idx].keys.selected[0]}"])
+    |> group()
+    |> distinct(column: "${cards[idx].keys.selected[0]}")${searchString}
+    |> limit(n: ${limit})
+    |> sort()`
+
+    if (
+      data.buckets[0].type !== 'sample' &&
+      isFlagEnabled('MeasurementFieldsNoTime')
+    ) {
+      _source = `import "regexp"
+      import "influxdata/influxdb/schema"`
+      queryText = `${_source}
+  schema.tagValues(
+    bucket: "${data.buckets[0].name}",
+    tag: "${cards[idx].keys.selected[0]}",
+    predicate: (r) => ${tagString},
+    start: ${CACHING_REQUIRED_START_DATE},
+    stop: ${CACHING_REQUIRED_END_DATE},
+  )${searchString}
+  |> group()
```

Comment:
Is this call to `group()` needed here? I think that the result of `tagValues` will already be grouped altogether. If for some reason it's needed, I'm curious what it is.

```suggestion
```

Reply:
👍